blog.md

```diff
@@ -0,0 +1,94 @@
+
+Before we start, we want to ensure we have the following libraries installed using pip install *package_name*: pandas, numpy, requests and bs4. The first thing we want to do for our scraper is determine how many pages of information there are. For this particular dataset, it is laid out across 9 different URLs. We have to examine what changes occur between two pages. It is a really long link, but we can see at the end there is a part that says offset=0 for the first page, but is equal to 100 for the second page, and increments by 100 each time. 
+
+Knowing what changes in our url, we will create a for loop that goes from 0 to 900 in increments of 100, and we will copy the base url (everything through `offset=`), and then add our for loop variable `i` to the end of the url with `str(i)`. So, `url = (base url) + str(i)`.
+
+![Alt Text](https://i.imgur.com/XQu2tug.png)
+
+For this dataset, I want to only extract the player, the season, their team, and their points per game. 
+Going back to our *td* pairs, I can click inspect element on each element and see what the corresponding *data-row=* is for each statistic. In HTML, they are tagged as 'player', 'season', 'team_id', 'pts_per_g'. We can create two Pandas dataframes to store our information: a temporary DataFrame we use to append a new page of information to our final set, and a final set to hold all the information.
+
+### Access Server Information
+
+At the start of each loop iteration, we are going to have to request an HTTP response for ourl url and store it as plain text. We then will pass it into a BeautifulSoup object, with the second parameter being a built in parser, cleverly named 'html.parser'. Our BeautifulSoup parser has two related functions to call: `find()` and `findAll()`. We can use `find()` to find the instance of our table. This is as opposed to using `findAll()`, when we need to search the entire document for instances of something, find returns the singular result we want directly rather than a list. For us, we are looking for up to 100 instances of relevant rows per page, so we want to use `findAll()`. We can create two for loops: one to loop over each relevant feature we want (player, season, etc.), and another one to loop over each *td* pair in our `findAll(*td*, *data-stat=feature*)`. Since we want the information as plain text, we have to convert it concurrently.
+
+![Alt Text](https://i.imgur.com/l5MaQso.png)
+
+# Append Data
+
+We now have our data, we want to re-format our data and add it to a final set. We take the stats variable and pass it into a Pandas DataFrame. Due to how the information was stored, our stats variable is a 4 element list with each element being size 100. We want the opposite of that, so we call `.transpose()` on our new dataframe to flip the rows to columns and vice versa, and set the column names to our list of features. We then call `.append()` with the final set DataFrame.
+
+# Bonus: Cleaning The Data
```

Comment:
re: conversation in our meeting, worth showing a screenshot pre- vs. post-filtered to demonstrate the filtering in this section. Otherwise, this section looks pretty good